tls/s2n_server_hello_retry.c

```diff
@@ -85,7 +85,7 @@ int s2n_server_hello_retry_recv(struct s2n_connection *conn)
     if (named_curve != NULL && s2n_ecc_preferences_includes_curve(ecc_pref, named_curve->iana_id)) {
         selected_group_in_supported_groups = true;
     }
-    if (kem_group != NULL && s2n_kem_preferences_includes_tls13_kem_group(kem_pref, kem_group->iana_id)) {
+    if (kem_group != NULL && kem_group->available && s2n_kem_preferences_includes_tls13_kem_group(kem_pref, kem_group->iana_id)) {
         selected_group_in_supported_groups = true;
```

Comment:
Should s2n_kem_preferences_includes_tls13_kem_group check kem_group->available for you?

Reply:
this is a good question, and one that i grappled with during a refacor during the last round of review. I initially implemented it this way, but @goatgoose [raised a keen suggestion](https://github.com/aws/s2n-tls/pull/4101#discussion_r1357146597) for how we could [simplify changes to `s2n_tls13_pq_handshake_test`](https://github.com/aws/s2n-tls/pull/4101/commits/983fb65803c0f9ad1f6433c1f5809fae773007c5), and I found that this additional check was a consequence of that.

Reply:
My thinking is that the other kem_preferences methods respect availability, so it might be surprising behavior if this one doesn't.

Reply:
there are [3 other call sites (not counting tests)](https://github.com/search?q=repo%3Aaws%2Fs2n-tls+s2n_kem_preferences_includes_tls13_kem_group&type=code) for `https://github.com/search?q=repo%3Aaws%2Fs2n-tls+s2n_kem_preferences_includes_tls13_kem_group&type=code`, and this is the only one that is concerned with availability. if we were to conflate availability with membership in a preference set, it would require us to rework a lot of the internal logic around/in those other 3 call sites.

to my mind, "is this available" and "is this item in that set" are distinct questions. is there a place we can document this distinction to mitigate potential surprise?